server/test/unit/server/managers/repo/test_publish.py

```diff
@@ -124,24 +124,26 @@ def test_publish_failure_report(self):
         mock_plugins.MOCK_DISTRIBUTOR.publish_repo.return_value = PublishReport(False, 'Summary of the publish', 'Details of the publish')
 
         # Test
-        report = self.publish_manager.publish('repo-1', 'dist-1', None)
+        try:
```

Comment:
with unittest2 (the PR for which hasn't been reviewed yet...), you can do basically this, which I think is a bit cleaner and more direct:

```
with self.assertRaises(PulpCodedException) as assertion:
  self.publish_manager.publish('repo-1', 'dist-1', None)
self.assertEqual(assertion.exception.error_code, error_codes.PLP0034)
```

No need to wait for unittest2, but it's worth knowing about.